exampleConnectionService/src/main/java/com/twilio/voice/examples/connectionservice/VoiceActivity.java

```diff
@@ -0,0 +1,337 @@
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.Snackbar;
+import android.support.design.widget.CoordinatorLayout;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.LocalBroadcastManager;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.telecom.DisconnectCause;
+import android.telecom.PhoneAccount;
+import android.telecom.PhoneAccountHandle;
+import android.telecom.TelecomManager;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.EditText;
+
+import com.koushikdutta.async.future.FutureCallback;
+import com.koushikdutta.ion.Ion;
+import com.twilio.voice.Call;
+import com.twilio.voice.CallException;
+import com.twilio.voice.LogLevel;
+import com.twilio.voice.LogModule;
+import com.twilio.voice.Voice;
+import com.twilio.voice.examples.connectionservice.R;
+
+import java.util.HashMap;
+
+public class VoiceActivity extends AppCompatActivity {
+    private static final String TAG = "VoiceActivity";
+
+    /*
+     * You must provide the URL to the publicly accessible Twilio access token server route
+     *
+     * For example: https://myurl.io/accessToken
+     */
+    private static final String TWILIO_ACCESS_TOKEN_SERVER_URL = "TWILIO_ACCESS_TOKEN_SERVER_URL";
+
+    public static final String OUTGOING_CALL_ADDRESS = "OUTGOING_CALL_ADDRESS";
+
+    public static final String ACTION_OUTGOING_CALL = "ACTION_OUTGOING_CALL";
+    public static final String ACTION_DISCONNECT_CALL = "ACTION_DISCONNECT_CALL";
+    public static final String ACTION_DTMF_SEND = "ACTION_DTMF_SEND";
+    public static final String DTMF = "DTMF";
+    public static final String CALLEE = "to";
+
+    private static final int MIC_PERMISSION_REQUEST_CODE = 1;
+    private static final int CALL_PHONE_CODE = 2;
+    private static final int SNACKBAR_DURATION = 4000;
+    private CoordinatorLayout coordinatorLayout;
+
+    private String accessToken;
+    private boolean isReceiverRegistered = false;
+    private VoiceBroadcastReceiver voiceBroadcastReceiver;
+
+    int PERMISSION_ALL = 1;
+    String[] PERMISSIONS = {Manifest.permission.RECORD_AUDIO, Manifest.permission.CALL_PHONE};
+
+    private TelecomManager telecomManager;
+    private PhoneAccountHandle handle;
+    private PhoneAccount phoneAccount;
+
+    HashMap<String, String> twiMLParams = new HashMap<>();
+    private AlertDialog alertDialog;
+    Call activeCall;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_voice);
+        coordinatorLayout = findViewById(R.id.coordinator_layout);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
```

Comment:
Use a better name for this R.id.

Reply:
Also the default icon is the icon used to disconnect a call. See quickstart behavior.